### PR TITLE
output moniker group in xrefmap to compress data

### PR DESCRIPTION
**Pull request by live1206**
[AB#411112](https://dev.azure.com/ceapex/Engineering/_workitems/edit/411112)

### CODE REVIEW

**Review by yufeih — APPROVED** [review]

File: src/docfx/build/xref/XrefResolver.cs [hunk 1]
```diff
@@ -158,14 +158,20 @@ public XrefMapModel ToXrefMapModel()
...
                 .ToArray();
 
-            externalXrefs = _jsonSchemaTransformer().GetValidateExternalXrefs();
+            var monikerGroups = new Dictionary<string, MonikerList>(
+                from item in references
+                let monikerGroup = item.MonikerGroup
+                where !string.IsNullOrEmpty(monikerGroup)
+                orderby monikerGroup
+                group item by monikerGroup into g
+                select new KeyValuePair<string, MonikerList>(g.Key, g.First().Monikers));
```
Comment (mingwli) [inline src/docfx/build/xref/XrefResolver.cs:167]: some minor thinking: can this be shared with publish model somehow?

Comment (live1206) [reply src/docfx/build/xref/XrefResolver.cs:167]: Thought about it, did not find a proper utility for it, though.

Comment (mingwli) [reply src/docfx/build/xref/XrefResolver.cs:167]: What about caching a the moniker map<hash, monikers> here https://github.com/dotnet/docfx/blob/v3/src/docfx/build/moniker/MonikerProvider.cs#L68-L71?

Comment (live1206) [reply src/docfx/build/xref/XrefResolver.cs:167]: Actually, there is not much to share in this case.
We have different items, both item happen to have `MonikerGroup` and `Monikers`, and what we are doing is to construct a mapping from `MonikerGroup` to `Monikers`, `MonikerGroup` is computed already. No extra calculation or cache needed.

**Pull request merged**